### PR TITLE
Require JDK 13 for compilation

### LINKED ISSUE #42895: Enable building with the JDK 13 compiler
We continually advance to the latest compiler as new versions of the JDK are released. However, we are prevented from trying to build with JDK 13 at this time since Gradle does not yet support JDK 13 (gradle/gradle#8681). This is a placeholder to enable building with the JDK 13 compiler when upstream addresses the issues that are preventing Gradle from supporting JDK 13.

Relates 18e74cb92e4cf977d5a117941d818e11ea35d031

### PR DESCRIPTION
**Pull request by jasontedor**
We have a long history of advancing the required compiler to the newest JDK. JDK 13 has been with us for awhile, but we were blocked from upgrading since Gradle was not compatible with JDK 13. With the advancement in our project to Gradle 6 which supports JDK 13, we can now advance our minimum compiler version. This commit updates the minimum compiler version to JDK 13.

Closes #42895

### CODE REVIEW

**Review by mark-vieira — APPROVED** [review]
LGTM, although we should probably update [this section of the contributing docs ](https://github.com/elastic/elasticsearch/blob/master/CONTRIBUTING.md#contributing-to-the-elasticsearch-codebase)to reflect folks will need a Java 13 JDK to build as well.

**Comment by mark-vieira** [comment]
Looking at the packaging test failures I'm not sure why Jenkins isn't picking up the change to `ES_BUILD_JAVA`. Perhaps something funny happened and it didn't see you second commit? Triggering again.

**Comment by mark-vieira** [comment]
@elasticmachine run elasticsearch-ci/packaging-sample-matrix

**Comment by pugnascotia** [comment]
Might be another case of https://github.com/elastic/infra/issues/15685

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -92,22 +92,22 @@ Contributing to the Elasticsearch codebase
...
 > Note: It is also required to have `JAVA8_HOME`, `JAVA9_HOME`, `JAVA10_HOME`
-and `JAVA11_HOME` available so that the tests can pass.
+and `JAVA11_HOME`, and `JAVA12_HOME` available so that the tests can pass.
 
 > Warning: do not use `sdkman` for Java installations which do not have proper
 `jrunscript` for jdk distributions.
 
 Elasticsearch uses the Gradle wrapper for its build. You can execute Gradle
-using the wrapper via the `gradlew` script on Unix systems or `gradlew.bat` 
+using the wrapper via the `gradlew` script on Unix systems or `gradlew.bat`
```
Comment (jasontedor) [inline CONTRIBUTING.md]:
```suggestion
using the wrapper via the `gradlew` script on Unix systems or `gradlew.bat` 
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -157,9 +157,9 @@ For IntelliJ, go to
 For Eclipse, go to `Preferences->Java->Installed JREs` and add `-ea` to
 `VM Arguments`.
 
-Some tests related to locale testing also require the flag 
+Some tests related to locale testing also require the flag
```
Comment (jasontedor) [inline CONTRIBUTING.md]:
```suggestion
Some tests related to locale testing also require the flag 
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -157,9 +157,9 @@ For IntelliJ, go to
...
 `-Djava.locale.providers` to be set. Set the VM options/VM arguments for
-IntelliJ or Eclipse like describe above to use 
+IntelliJ or Eclipse like describe above to use
```
Comment (jasontedor) [inline CONTRIBUTING.md]:
```suggestion
IntelliJ or Eclipse like describe above to use 
```

**Review by pugnascotia — APPROVED** [review]

**Pull request merged**